File: Source/Ivxr.SpaceEngineers/WorldModel/CharacterMovementFlags.cs
using System;

namespace Iv4xr.SpaceEngineers.WorldModel
{
    [Flags]
    public enum CharacterMovementFlags : byte
    {
        Jump = 1,
        Sprint = 2,
        FlyUp = 4,
        FlyDown = 8,
        Crouch = 16,
        Walk = 32
    }
}

File: JvmClient/src/commonMain/kotlin/spaceEngineers/model/CharacterMovementFlags.kt
package spaceEngineers.model

import kotlinx.serialization.Serializable
import kotlin.jvm.JvmInline

@Serializable
@JvmInline
value class CharacterMovementFlags(val value: UShort) {

    val wantsWalk: Boolean
        get() = value and Walk.toUShort() > 0u

    companion object {
        val Jump = 1
        val Sprint = 2
        val FlyUp = 4
        val FlyDown = 8
        val Crouch = 16
        val Walk = 32
    }
}
